jcl/src/java.base/share/classes/openj9/management/internal/ThreadInfoBase.java

```diff
@@ -238,19 +238,39 @@ public boolean equals(Object obj) {
 		return result;
 	}
 
+	private int hashCodeValue = 0;
 	/**
 	 * {@inheritDoc}
 	 */
 	@Override
 	public int hashCode() {
-		return (Long.toString(blockedCount) + blockedTime + lockName + lockOwnerId
-				+ lockOwnerName + stackTrace.length + threadId + threadName
-				+ threadState + waitedCount + waitedTime + inNative
-				+ suspended
-/*[IF Sidecar19-SE]*/
-				+ daemon + priority
-/*[ENDIF]*/
-		).hashCode();
+		/* Ignore the rare case when the hashcode evaluates to 0,
+		 * since the cost of re-evaluating
+		 * the code is less than the cost of an extra boolean field in the object.
+		 */
+		if (0 == hashCode()) {
```

Comment:
Do you mean `hashCodeValue` instead of `hashCode()`? Seems this is recursive.

Reply:
True.  Fixed and pushed.  Tested manually:

```
package experimental11;

import java.lang.Thread.State;

import openj9.management.internal.ThreadInfoBase;

public class hashcode {

	@SuppressWarnings("nls")
	public static void main(String[] args) {
		StackTraceElement[] trc = Thread.currentThread().getStackTrace();
		for (int i = 0; i < 100; i++) {
			ThreadInfoBase tib = new ThreadInfoBase(i, "foo" + i, Thread.State.RUNNABLE, intToBool(i, 2),
					intToBool(i, 4), i * 3, i * 11, i * 13, i * 17, "bar" + i, i * 19, "toe" + i, trc, null, null, null,
					intToBool(i, 6), (i * 23) % 8);
			final int hc1 = tib.hashCode();
			final int hc2 = hc1;
			System.err.println(hc2 + " " + hc2 + ": " + ((hc1 == hc2) ? "match" : "mismatch"));
		}
	}

	static boolean intToBool(int i, int mod) {
		return 1 == ((i % mod) & 1);
	}
}
```